src/app/route-history/route-history.service.ts

```diff
@@ -11,12 +11,19 @@ export class RouteHistoryService implements OnDestroy {
   public previousRoute: string;
 
   private subscription: Subscription;
+  private popstate: boolean = false;
 
   constructor(router: Router) {
     this.subscription = router.events.subscribe((event: Event) => {
+      if (event instanceof NavigationStart) {
+        if (event.navigationTrigger === 'popstate') {
+          this.popstate = true;
```

Comment:
codecov is pointing out this isn't covered by a test -- could it be or is that a big lift?

Reply:
I tried to figure out how to cover this with a test and from what I saw it seemed impossible to actually simulate a `popstate` event with Angular's test router. I can take a look at it again though.

Reply:
If you looked already that's good enough for me!